tool/formula/src/main/java/org/openscience/cdk/tools/manipulator/MolecularFormulaManipulator.java

```diff
@@ -209,14 +209,31 @@ public static String getString(IMolecularFormula formula, String[] orderElements
             if (!elemSet.contains(symbol)) elemSet.add(symbol);
         }
 
-        for (String elem : elemSet) {
-            int count = 0;
-            for (IIsotope isotope : formula.isotopes()) {
-                if (isotope.getSymbol().equals(elem)) count += formula.getIsotopeCount(isotope);
+        if (!setMassNumber) {
+            for (String elem : elemSet) {
+                int count = 0;
+                for (IIsotope isotope : formula.isotopes()) {
+                    if (isotope.getSymbol().equals(elem)) count += formula.getIsotopeCount(isotope);
+                }
+                stringMF.append(elem);
+                if (!(count == 1 && !setOne)) stringMF.append(count);
+            }
+        } else {
+            for (IIsotope isotope : isotopesList) {
+                int count = formula.getIsotopeCount(isotope);
+                try {
+                    IIsotope major = Isotopes.getInstance().getMajorIsotope(isotope.getSymbol());
```

Comment:
This is wrong, the test should now be **isotope.getMassNumber() == null**. It is valid to have an input like ``[12CH3]CO`` as different from ``CCO``.